packages/server/utils/analytics/analytics.ts

```diff
@@ -0,0 +1,101 @@
+import {createMeetingTemplateAnalyticsParams} from './helpers'
+import {SegmentAnalytics} from './segment/SegmentAnalytics'
+
+export type AnalyticsEvent = 'Meeting Started' | 'Meeting Joined' | 'Meeting Completed'
+
+/**
+ * Provides a unified inteface for sending all the analytics events
+ */
+class Analytics {
+  private segmentAnalytics: SegmentAnalytics
+
+  constructor() {
+    this.segmentAnalytics = new SegmentAnalytics(segment)
+  }
+
+  teamPromptEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    responses: TeamPromptResponse[]
+  ) => {
+    const userIdsResponses: Record<string, string> = responses.reduce(
+      (previous, response) => ({...previous, [response.userId]: response.plaintextContent}),
+      {}
+    )
+    meetingMembers.forEach((meetingMember) => {
+      const plaintextResponseContent = userIdsResponses[meetingMember.userId]
+      this.meetingEnd(meetingMember.userId, completedMeeting, meetingMembers, {
+        responseAdded: !!plaintextResponseContent
+      })
+    })
+  }
+
+  checkInEnd = (completedMeeting: Meeting, meetingMembers: MeetingMember[]) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(meetingMember.userId, completedMeeting, meetingMembers)
+    )
+  }
+
+  retrospectiveEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    template: MeetingTemplate
+  ) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(
+        meetingMember.userId,
+        completedMeeting,
+        meetingMembers,
+        createMeetingTemplateAnalyticsParams(template)
+      )
+    )
+  }
+
+  sprintPokerEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    template: MeetingTemplate
+  ) => {
+    meetingMembers.forEach((meetingMember) =>
+      this.meetingEnd(
+        meetingMember.userId,
+        completedMeeting,
+        meetingMembers,
+        createMeetingTemplateAnalyticsParams(template)
+      )
+    )
+  }
+
+  private meetingEnd = (
+    userId: string,
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    meetingSpecificProperties?: any
+  ) => {
+    const {facilitatorUserId, meetingNumber, meetingType, phases, teamId} = completedMeeting
+    const presentMemberUserIds = meetingMembers.map(({userId}) => userId)
+    const wasFacilitator = userId === facilitatorUserId
+    this.track(userId, 'Meeting Completed', {
+      hasIcebreaker: phases[0]?.phaseType === 'checkin',
+      // include wasFacilitator as a flag to handle 1 per meeting
+      wasFacilitator,
+      userIds: wasFacilitator ? presentMemberUserIds : undefined,
+      meetingType,
+      meetingNumber,
+      teamMembersCount: meetingMembers.length, // TODO: use team members count
+      teamMembersPresentCount: meetingMembers.length,
+      teamId,
+      ...meetingSpecificProperties
+    })
+  }
+
+  private track = (userId: string, event: AnalyticsEvent, properties?: any) =>
```

Comment:
+1 My 2 cents: not having a generic method for tracking one of a type events will just add more boilerplate. I think for meetings it is fine to do the formatting here, but if I look at inviteToTeam, createTaskIntegration etc. these are easier to handle locally. I'm fine defining the event type here though so there is some place to keep an overview.

Reply:
yes, I understand the tension. it's a bit of a personal preference, I prefer more readable code even with the cost of more boilerplate. we have a choice of either using `analytics.track(userId, 'Invite', { teamId, invitees: successfulInvitees})` or `analytics.inviteEmailSent(userId, invites)`. The reason I prefer the second is that I think it's a bit better to delegate the formatting of the parameters to the class that's actually responsible for successfully handling that. the less we have to change the resolvers when dealing with their side effects, the better IMO. I'm open to making that public though, it'd still be an improvement over what we do now.